app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -203,9 +207,8 @@ class ConversationListViewModel @Inject constructor(
         state = state.copy(blockUserDialogSate = BlockUserDialogState(name, id))
     }
 
-    // TODO: needs to be implemented
-    @Suppress("EmptyFunctionBlock")
-    fun leaveGroup(id: String = "") {
+    fun leaveGroup(conversationId: ConversationId) = viewModelScope.launch(dispatchers.io()) {
+        leaveGroupConversation(conversationId = conversationId, selfUserId)
```

Comment:
I was wondering if we could get this info in the `UseCase` rather than passing it as parameter, that could simplify the flow, at least we use the `selfUser` for other purposes in the view for example

Reply:
We totally should inject that in the use case not the viewModel

Reply:
But this use case is called `removeMemberFromConversation` so it is designed in such a way that the `userId` needs to be specified in advance. That way we will be able to reuse the same use case for a user that decides to leave voluntarily the given conversation as well as invoking it when an admin decides to remove a user from a conversation he's part of.

Reply:
A different approach would be to have 2 different use cases, `leaveGroupUseCase` where the `selfUserId` would be directly injected inside, and another `removeMemberFromConversationUseCase` where we will provide the userId of the user we want to remove. 
But imho it would be a bit overkill as both use cases will be implemented in the same way internally.
 What do you think @MohamadJaara ?

Reply:
IMO, I agree with you, I'm ok to take the pragmatic approach and just leave it as it is right now and pass the identifier as an argument. 🥘 💯